lib/momento/cache_client.rb

```diff
@@ -455,16 +439,18 @@ def to_bytes(string)
       end
     end
 
-    # This is not a complete validation of the cache name, just
-    # issues that might cause an exception in the client. Let the server
-    # handle the rest of the validation.
+    # Return a UTF-8 version of the cache name.
     #
     # @param name [String] the cache name to validate
     # @raise [TypeError] when the name is not a String
-    # @raise [Momento::CacheNameError] when the name is not ASCII
+    # @raise [Momento::CacheNameError] when the name is not UTF-8 compatible
     def validate_cache_name(name)
       raise TypeError, "Cache name must be a String, got a #{name.class}" unless name.is_a?(String)
-      raise Momento::CacheNameError, "Cache name must be ASCII, got '#{name}'" if name.match?(/[^[:ascii:]]/)
+
+      encoded_name = name.encode('UTF-8')
```

Comment:
I thought this would raise an error, instead it returns the string [Invalid encoding] with `valid_encoding?` equal to false.